src/vm/gchelpers.cpp

```diff
@@ -1618,3 +1618,22 @@ SetCardsAfterBulkCopy(Object **start, size_t len)
 #if defined(_MSC_VER) && defined(_TARGET_X86_)
 #pragma optimize("", on)        // Go back to command line default optimizations
 #endif //_MSC_VER && _TARGET_X86_
+
+INT64 GetAllocatedBytesForCurrentThread()
+{
+    INT64 currentAllocated;
+
+    if (GCHeapUtilities::UseThreadAllocationContexts())
+    {
+        gc_alloc_context* ac = GetThread()->GetAllocContext();
+        currentAllocated = ac->alloc_bytes + ac->alloc_bytes_loh - (ac->alloc_limit - ac->alloc_ptr);
+    }
+    else
+    {
+        GlobalAllocLockHolder holder(&g_global_alloc_lock);
```

Comment:
This will return allocations from all threads. I am not sure whether it is an improvement compared to current behavior.

Maybe we should rather make this return `-1` when the runtime does not support per-thread allocation counters, and then folks can choose to either use the global allocation counter or something else?